common/src/device/transport/mqtt/ClientWebSocketChannel.cs

```diff
@@ -96,6 +96,11 @@ protected override void DoDisconnect()
         }
 
         protected override async void DoClose()
+        {
+            await DoCloseAsync().ConfigureAwait(false);
```

Comment:
This won't change much: the async void isn't awaited and if DoClose throws it will still be unobserved.

Reply:
@CIPop I agree, and I think the issue cannot be fixed in this repository alone as the underlying library called DoNetty (https://github.com/Azure/DotNetty) needs to be fixed first. I took a crack at it but soon realized it is bigger task than i anticipated or can afford to spend time to fix. Basically the IChannel and all its hierarchy needs to be converted to return Task instead of void. That change should be done all the way up and ideally with name changes to methods such as Open to OpenAsync. I wish there was a short cut but it does look like it.

Reply:
@SwatantraYadav I agree - this is definitely non-trivial and requires a lot of investigations to ensure all tasks get properly awaited or at least results get observed. 
We work closely with the DotNetty team and we will raise this up for a future release.